consistency/tool/comparison/comparison.py

```diff
@@ -45,19 +40,39 @@ def find_closest_id(word):
             closest_word_groups[word] = comparison_info
         return closest_word_groups
 
-    def find_word_group(self):
-        # TODO: using the map returned by find_word_group,
-        # if the vocab has an associated ID, form a Variation object
-        # add this object into the current_variations list
-        # if not, check if the vocab is a NOISE_WORD, if not, add it the
-        # unique_terms list object. If there is a matching word, form a variation object
-        # from the match -> means creating a wordgroup instance at the dot.
-        # return the current variations object
-        return
-
-    def find_past_variations(self):
-        # TODO: check the timestamp for the uploaded spec(s) and compare
-        #  against the timestamp of the specs we used in the clustering step.
-        # if different the variations we last computed are outdated as such are no longer current.
-        # we recompute the comparison to get current variations
-        return
+    # format: {word : [wordgroup, distance]}
+    def generate_consistency_report(self):
+
+        if self.word_groups == None:
+            return None
+
+        closest_word_groups = self.find_closest_word_groups()
+        report = cr.ConsistencyReport()
+        current_variations = []
+        past_variations = []
+        unique_words = []
+
+        for word in closest_word_groups:
+            if closest_word_groups[word][0] == self.noise_words and word not in self.noise_words:
+                unique_words.append(word)
+            else:
+                if word not in list(closest_word_groups[word][0].word_frequency):
```

Comment:
Any specific reason to convert `word_frequency` into a list? `word_frequency` is a map in the original proto which will help with faster lookup.